Rename to Luanti

grorp opened the pull request:
This PR does the most fundamental part of the rename: The CMake-provided project name, used in various places, is changed to Luanti. I wanted to make a tiny PR first, but there's no sane way to change the project name without also changing lots of other things. Notes:

- Executable file name is changed
- Locale file names are changed
- path_share & path_locale are changed
- path_user & path_cache are unchanged for backwards compat, would need a migration
- minetest.conf file name is unchanged
- net.minetest.minetest ID is unchanged on Android and Linux (cannot be changed on Android), changed to org.luanti.luanti on macOS

There's also a new header since the title bar saying "Luanti" and the header saying "MINETEST" felt a little too inconsistent. The new header intentionally isn't in MTG stone design (or any other MTG-esque design), since we want to distance Luanti and MTG from each other. The font [undefined medium](https://undefined-medium.com/) was used. Disclaimer: I'm not an artist.

Screenshots:
<img alt="mainmenu" src="https://github.com/user-attachments/assets/17175811-8ea7-46c8-8458-c75475b258fb" width="512" />
<img alt="in-game" src="https://github.com/user-attachments/assets/6da3bcc5-d1b6-4644-83b4-2c6ad4bb6f03" width="512" />

There's also a new ASCII splash, using the same font as the old one.

## To do

This PR is a Ready for Review.

Reviewing is easier if you ignore the unchanged renamed .po files

## How to test

Compile and look. Test that your worlds etc. are still found. Test that translations work.

- [x] Linux RUN_IN_PLACE
- [x] Linux make install
- [x] Android
- [x] Windows
- [x] macOS

sfan5 commented on src/CMakeLists.txt:
```diff
@@ -1,17 +1,17 @@
-project(minetest)
+project(luanti)
```
isn't this redundant actually?

grorp replied:
e0b6668d6f711fcc51e3e81be4a4781aead9f7ac

sfan5 replied:
i guess that's for another day :shrug: 
because we already have a `project()` call in the top-level file

sfan5 commented:
what about a `minetest` and `mineteserver` wrapper executable? (at least for Unix)

> Locale file names are changed

will this work with Weblate? or does it need special attention?
at the very least any pending commits would not apply correctly

sfan5 commented on misc/winresource.rc:
```diff
@@ -49,8 +49,7 @@ BEGIN
         VALUE "InternalName", PROJECT_NAME
-        VALUE "LegalCopyright", "(c) 2011-2015 celeron55"
-        VALUE "LegalTrademarks", """Minetest"" is the property of the Minetest community, don't use it without permission!"
+        VALUE "LegalCopyright", "(c) 2010-2024 celeron55 and contributors"
```
fwiw it might be more legally valid to use c55's full name

grorp commented:
> what about a `minetest` and `mineteserver` wrapper executable? (at least for Unix)

How do you do this? Symlink? Shell script that prints a warning before launching the new executable? Something else?

> will this work with Weblate? or does it need special attention?
> at the very least any pending commits would not apply correctly

No idea. I renamed the .po files because they're referenced by PROJECT_NAME in various places and making them continue to use "minetest" would have been additional work, not less work.

grorp commented:
> How do you do this? Symlink? Shell script that prints a warning before launching the new executable? Something else?

Shell script alias with deprecation warning implemented in 5e48ca2b8c9c1eccb0921cd330e2aaa1139f7d37.

grorp commented:
> I'm probably a fan of the first and, if the L right column is removed, of the second. Thoughts?

The first could be fine, can you make a `menu_header.png` out of it?

sfan5 commented:
> How do you do this?

I was thinking shell script.
But symlink + checking `argv[0]` in main.cpp is a more portable approach.

> (And it needs to be done anyway at some point.)

I don't disagree. Just saying if it causes too much trouble before release we could skip that for now.

sfan5 commented on misc/aliases/minetest:
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+echo -e "\033[93mminetest is a deprecated alias for luanti\033[0m"
+dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+"$dir/luanti" "$@"
```
for completeness this would have been my complaints:
* requiring bash is probably not a good idea (not portable)
* `echo -e` is unportable
* printing color codes without regard for tty
* finding the script dir is probably unnecessary since we can assume the user has both in his PATH?
* missing `exec`

grorp commented:
@sfence can you please test this PR on macOS?

celeron55 commented:
> A few font proposals
> 
> ![image](https://private-user-images.githubusercontent.com/63455151/377063727-5ede80dc-a52a-4790-87d1-f6867eac0f3e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkxNjUzODQsIm5iZiI6MTcyOTE2NTA4NCwicGF0aCI6Ii82MzQ1NTE1MS8zNzcwNjM3MjctNWVkZTgwZGMtYTUyYS00NzkwLTg3ZDEtZjY4NjdlYWMwZjNlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDE3VDExMzgwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZhYWUwZjMzYzlhN2MxZjA4MTcxNzNkMTc2ODFlZTBjMmZhN2Y3YjUzYjVmNjY0MWNlMzA2NzhjZjFhZTM4NzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Hpq-QNxkg_ifmNg61TYSapNRvs86tjXESjfuyrQTL9k)
> 
> I'm probably a fan of the first and, if the L right column is removed, of the second. Thoughts?

For what it's worth I liked the very simple pixelated font that this PR originally showed up with. It's not in that picture. I think all of these are worse.

SwissalpS commented:
> A few font proposals
> Thoughts?

I agree with celeron55, the first screenshot looks better.
As for the ones in second proposal, the first line looks best but would need to have the spaces between L and U and U and A smaller so the U doesn't float around as much.

grorp commented:
Switched back to the previous header image.

I would like to note that the header image is the simplest part of this PR. It's just an image file and can be changed at any time by future PRs. The technical decisions made here are much more important.

Desour reviewed (commented):
Tested.
Found no bigger issues.

But idk enough about all the stuff here to properly judge, aka I don't feel qualified enough to approve.

Zughy commented on textures/base/pack/menu_header.png:
This is the recipe for bikeshedding 101. If we want a professional logo, we can't have it in a couple of days, so further discussion about it is only a waste of time. Let's please focus on actually shipping something that doesn't break

Desour replied:
Agreed, this is why I hid my suggestion in a review. :D

chmodsayshello commented:
Just want to mention that there still is the ascii art that says "minetest" whenever a server is started in https://github.com/grorp/minetest/blob/the-root-of-renaming/src/server.cpp#L588

I suggest replacing it with
```
.__                       __  .__ 
|  |  __ _______    _____/  |_|__|
|  | |  |  \__  \  /    \   __\  |
|  |_|  |  // __ \|   |  \  | |  |
|____/____/(____  /___|  /__| |__|
                \/     \/         
```

SwissalpS commented:
IMO first one for now

grorp commented:
> It's probably worth contacting downstream/distro maintainers about how they'd handle a rename, especially with the executable name changing

Interesting: https://wiki.debian.org/RenamingPackages

Gamer-Kold commented:
```
  _           _      _    _______     _______  
 / |         / |    | |  /   _/  \   /   _/  \
|  |        |  |    | | |    _  _ | |         |
| /|        |  |    | | |/  / \  \| |/  ____  |
|  |        |  |    | | |   \_/   | |  |    | |
|  |        |_/|    | | |_/ ____  | |  |    | |
|_/|_______ |  |____| | |  |    | | |_/|    | |
|     /   / |    _/   | |  |    | | |  |    | |
 \_______/   \_______/   \_|    |_|  \_|    |_|
```
Here's a very work in progress proposal for the ascii banner. It's handmade so it might take some time to polish up. I was going for carved out of stone look. How do we feel about it so far?

sfan5 commented:
I think the very first of @chmodsayshello's suggestion is best, because it matches the existing style.
Also as mentioned the technical decisions are more important here. We can change art later.

Gamer-Kold commented:
Right.

sfence commented:
Testing on MacOS:
Build works.
Build output also works.

CI generated Luanti: No problem found.

swagtoy commented:
I'd rather see a handmade Luanti logo as @Gamer-Kold is doing. The previous ones are very obviously made with figlet and it seems lazy.

swagtoy commented:
gotcha i didn't see that

sfence reviewed (changes requested):
We have in source files:

```
/*
Minetest
Copyright (C) 20....
```
Don't we want to change at least in the changed files?

`src/settings_translation_file.cpp `-> multiple `gettext` functions use Minetest in string.

`builtin/mainmenu/*` -> Multiple files using `fgettext` with Minetest in string.
Also `credit.json` in `mainmenu` use the old name.

What about some mod descriptions etc in devtest? Do we want to change them?

`irr/README.md` -> URL to Minetest. If we migrate the git project with the release, will be unactual.

`Licence.txt `-> Replace `Minetest` with `Luanti`?

`android/app/bild.gradle` -> Minetest.zip ?

What about `po` directory? Do we want to keep Minetest here in translation texts?

grorp commented:
> We have in source files:
> 
> ```
> /*
> Minetest
> Copyright (C) 20....
> ```
> 
> Don't we want to change at least in the changed files?

No, the copyright headers should be updated all at once if we decide to apply the rename to them. Ideally, this would happen together with #14439. This is out of scope for this PR.

> `src/settings_translation_file.cpp `-> multiple `gettext` functions use Minetest in string.

This is a generated file, updating it manually wouldn't make sense. Changes will start propagating once source strings are edited.

> `builtin/mainmenu/*` -> Multiple files using `fgettext` with Minetest in string. Also `credit.json` in `mainmenu` use the old name.

I've updated a few more of these, I consider the rest out of scope.

> What about some mod descriptions etc in devtest? Do we want to change them?
> 
> `irr/README.md` -> URL to Minetest. If we migrate the git project with the release, will be unactual.
> 
> `Licence.txt `-> Replace `Minetest` with `Luanti`?

These are for the documentation PR or future documentation PRs, out of scope for this PR.
 
> `android/app/bild.gradle` -> Minetest.zip ?

I've renamed that one to `assets.zip`.

> What about `po` directory? Do we want to keep Minetest here in translation texts?

These are generated files, updating them manually wouldn't make sense. Changes will start propagating once source strings are edited.

-----------

Please note that this is not a generic "replace all occurences of the old name with the new name" PR. This PR is for renaming the Minetest app/program to Luanti.

sfan5 commented on textures/base/pack/menu_header.png:
why all the empty space? won't it be centered anyway?

grorp replied:
Without the horizontal padding, the header would be huge:

![screenshot](https://github.com/user-attachments/assets/2b7bbead-49ff-4cef-862f-f0dbeddea0ef)

There's no vertical padding because the header is already tiny on Android and vertical padding would make it worse

sfence reviewed (approved):
Let's go and fix other things in other PRs.

Wuzzy2 commented:
Please don't pick an ASCII art that is all-lowercase. Because that looks like "Iuanti".

Zughy commented:
@sfan5 any updates for the review?

Wuzzy2 commented:
Tested 246d60df0f6807ad0218dbe8bf9c45ef328ec053 and I can confirm:

* That it compiles (clean, fresh repository)
* That the German translation works (both engine + mod TR files)
* That launching a singleplayer game works (using an existing `~/.minetest` data directory)
* That in-game texts say "Luanti" now (except in some settings tooltips because that's done in #15324)
* That the window header, Esc menu, debug screen and About tab say "Luanti" now
* That the ASCII art works
* That the new text logo in the main menu works

Looks good so far. :+1:

What this PR does not touch (these may or may not be out of scope):

* The `README` mentions some Minetest directories and environment vars in the `Paths` section (probably relevant since this PR is about renaming the project?)
* "Minetest" in code file headers (comments)
* Two C++ `#error`s in `main.cpp` mention Minetest
* Most builtin stuff (handled in PR #15324)

sfan5 reviewed (approved):
works.

grorp commented:
> Looks good so far. 👍

Thanks for testing.

> The `README` mentions some Minetest directories and environment vars in the `Paths` section (probably relevant since this PR is about renaming the project?)

I haven't applied the rename to these paths and environment variables and I don't plan to do that in this PR, since they would need additional code and testing for backwards compatibility.

> "Minetest" in code file headers (comments)

Out of scope for this PR, but there's already a different one: #15321

> Two C++ `#error`s in `main.cpp` mention Minetest

Fixed (looks like I can't use PROJECT_NAME_C in the #error message, it won't expand)

TailsFanLOL commented:
Please note what I said above is just a personal opinion, if you think it's "absurd to have the audacity to say this while having zero contributions", just ignore it.

TailsFanLOL commented:
@appgurueu, where would be the right place to discuss the description?

My main point currently is that there are many things "voxel" can refer to (and "roughly 1 cubic meter sized voxel without much physics for representing terrain" is too long for a description) and that it isn't really a game engine, the same way Roblox is referred to as a game or a game platform and not a game engine. Again, look at Teardown: it's a voxel game, yet it looks and functions nothing like games many would refer to as "minecrafty". Many are also unfamiliar with the definition term "voxel" and might think it's something "retro style" or "high-tech".
However, Roblox is pretty different I would say: instead of games there are "experiences" which aren't downloaded but rather "streamed" to the client, is multiplayer only and is forcing developers to include certain elements (eg. there's no way for a dev to remove the item system or the chat filter (anymore)).
